port_ocean/core/defaults/initialize.py

```diff
@@ -50,18 +53,52 @@ def deconstruct_blueprints_to_creation_steps(
     )
 
 
-async def _create_resources(
+async def _initialize_required_integration_settings(
     port_client: PortClient,
-    defaults: Defaults,
+    default_mapping: PortAppConfig,
     integration_config: IntegrationConfiguration,
 ) -> None:
-    response = await port_client._get_current_integration()
-    if response.status_code == status.HTTP_404_NOT_FOUND:
-        logger.info("Integration doesn't exist, creating new integration")
-    else:
-        logger.info("Integration already exists, skipping integration creation...")
-        return
+    try:
+        logger.info("Initializing integration at port")
+        integration = await port_client.get_current_integration(
+            should_log=False, should_raise=False
+        )
+        if not integration:
+            integration = await port_client.create_integration(
+                integration_config.integration.type,
+                integration_config.event_listener.to_request(),
+                port_app_config=default_mapping,
+            )
+        elif not integration["config"]:
+            integration = await port_client.patch_integration(
```

Comment:
add log about each case, indicating what you got and what you are about to apply